### ISSUE TITLE
Virtualization problem

### ISSUE THREAD

**Issue by jkopstals** [rcasero/doc#8]
Any ideas about grub console being displayed when launching the virtualized Windows machine?
I have exactly the same partition layout as you (windows/dell firmware uses first 6 partitions /dev/nvme0n1p1 .. p6, Ubuntu is on /dev/nvme0n1p7).
When launching the virtual machine, grub command line (grub> ) comes up, and windows does not start loading.
Any ideas would be appriciated :)

**Comment by rcasero** [comment]
I'm a bit confused. Are you running linux, and then from linux you launch VirtualBox to run Windows? Did you follow the instructions in 

https://github.com/rcasero/doc/wiki/Ubuntu-linux-on-Dell-XPS-15-(9560)#run-your-native-windows-10-pro-install-as-guest-on-virtualbox-with-ubuntu-as-host

?

**Comment by jkopstals** [comment]
Hi!
Yep - running Ubuntu 17.10. I would like to run Windows (guest) in VirtualBox from Ubuntu (host).
I followed the guide, but installed a fresh Ubuntu 17.10, instead of 16.04.

Virtualbox version Version 5.1.30_Ubuntu (current version for this ubuntu version, by running "sudo apt-get install virtualbox virtualbox-ext-pack").

Created rawdisk with:
sudo VBoxManage internalcommands createrawvmdk \
-filename "/home/jk/VirtualBox_VMs/win10pro.vmdk" \
-rawdisk /dev/nvme0n1 \
-partitions 1,2,3,4,5,6

otherwise I followed all the other steps as in your guide. Could the difference be that in ubuntu installation I chose to put the bootloader on the /dev/nvme0n1 (not a specific partition?). Not sure if that might be the issue in my case? Although, if I understand correctly that is the correct place to put the bootloader.

Anyhow, I'm currently getting the "grub>" command line when launching Windows in vbox. Windows still launches fine if run it natively, so the bootloader seems to be fine.

**Comment by rcasero** [comment]
@jkopstals I think I had a similar issue when I (disastrously) upgraded to 17.10. In my case, I fixed it by giving VirtualBox access to the whole disk, and not only the Windows partitions. However, this is very dangerous, because if by mistake you boot into linux (guest) from linux (host), the disk could get corrupted.

I added a note in the wiki on how to do it:

https://github.com/rcasero/doc/wiki/Ubuntu-linux-on-Dell-XPS-15-(9560)#letting-virtualbox-boot-the-native-windows-10-pro-installation-as-a-virtual-machine

> Note: If this causes the machine to not boot and instead go to the grub2 shell, you may need to allow access to the whole drive. This is very dangerous, because it will allow you to boot your Ubuntu host as a guest too, with both systems trying to access the same drive. If you do this, you need to make sure every time you boot your virtual machine, that you choose Windows.


```
sudo VBoxManage internalcommands createrawvmdk \
-filename "$HOME/VirtualBox_VMs/win10pro.vmdk" \
-rawdisk /dev/nvme0n1
```



A better solution may be the fix reported by @FloEdelmann in issue #12 , but I haven't tested it yet.

My advice would be: back up, then try @FloEdelmann 's solution, and please report back in issue #12 (I'll close this issue). If that doesn't work, and you don't mind risking your whole drive, then you can try what I did.
